lms/djangoapps/courseware/masquerade.py

```diff
@@ -19,16 +25,21 @@
 # The value is a dict from course keys to CourseMasquerade objects.
 MASQUERADE_SETTINGS_KEY = 'masquerade_settings'
 
+# The key used to store temporary XBlock field data in the Django session.  This is where field
+# data is stored to avoid modifying the state of the user we are masquerading as.
+MASQUERADE_DATA_KEY = 'masquerade_data'
+
 
 class CourseMasquerade(object):
     """
     Masquerade settings for a particular course.
     """
-    def __init__(self, course_key, role='student', user_partition_id=None, group_id=None):
+    def __init__(self, course_key, role='student', user_partition_id=None, group_id=None, user_name=None):
         self.course_key = course_key
         self.role = role
         self.user_partition_id = user_partition_id
         self.group_id = group_id
+        self.user_name = user_name
```

Comment:
What happens if `role` and `user_name` are inconsistent? e.g. `staff` with a student username or `student` with a staff username? (the latter is certainly possible to trigger from the code from what I can see). Shouldn't that be validated?